hack/kind-up.sh

```diff
@@ -110,3 +110,16 @@ kubectl apply -k "$(dirname "$0")/../example/gardener-local/metrics-server"   --
 kubectl get nodes -l node-role.kubernetes.io/control-plane -o name |\
   cut -d/ -f2 |\
   xargs -I {} kubectl taint node {} node-role.kubernetes.io/master:NoSchedule- node-role.kubernetes.io/control-plane:NoSchedule- || true
+
+# Allow multiple shoot worker nodes with calico as shoot CNI: As we run overlay in overlay ip-in-ip needs to be allowed in the workload.
+# Unfortunately, the felix configuration is created on the fly by calico. Hence, we need to poll until kubectl wait for new resources
+# (https://github.com/kubernetes/kubernetes/issues/83242) is fixed. (2 minutes should be enough for the felix configuration to be created.)
+for ((i = 0; i < 120; i++)); do
+  if ! kubectl get felixconfiguration default; then
+    echo "FelixConfiguration 'default' not found, yet, waiting..."
+    sleep 1s
+  else
+    kubectl patch felixconfiguration default --type merge --patch '{"spec":{"allowIPIPPacketsFromWorkloads":true}}'
+    break
+  fi
+done
```

Comment:
```suggestion
felix_config_found=0
max_retries=3
for ((i = 0; i < max_retries; i++)); do
  if kubectl get felixconfiguration default >/dev/null 2>&1; then
    if kubectl patch felixconfiguration default --type merge --patch '{"spec":{"allowIPIPPacketsFromWorkloads":true}}' >/dev/null 2>&1; then
      echo "FelixConfiguration 'default' successfully updated."
      felix_config_found=1
      break
    fi
  fi
  echo "retrying"
  sleep 1
done

if [ $felix_config_found -eq 0 ]; then
  echo "Error: FelixConfiguration 'default' not found or patch failed after $max_retries attempts."
  exit 1
fi
```
WDYT ?

Reply:
In a local test it took 15 seconds until the felix configuration was available. Therefore, I do not see `3` as a valid option for maximum number of retries.
Apart from that, I have no problem in changing the structure, e.g. redirecting the output to `/dev/null`. However, just printing `retrying` several times might be a bit confusing.

What do you think about the following?

```
for ((i = 0; i < 120; i++)); do
  if ! kubectl get felixconfiguration default > /dev/null 2>&1; then
    echo "FelixConfiguration 'default' not found, yet, retrying in 1s..."
    sleep 1s
  else
    kubectl patch felixconfiguration default --type merge --patch '{"spec":{"allowIPIPPacketsFromWorkloads":true}}'
    break
  fi
done
```

Reply:
It was my mistake that the suggested script contained my debugging logs `echo "retrying"` and wrong max retries `3`. I have now updated the suggestion. 
However, my concern is that we should exit after reaching the maximum retries and retry if there is any error with the patch.

Reply:
Addressed with https://github.com/gardener/gardener/pull/7684/commits/2247f140f9c235638306f6a81916b5247d337f8e.

WDYT?